projects/opendr_ws_2/src/data_generation/data_generation/synthetic_facial_generation.py

```diff
@@ -0,0 +1,166 @@
+                                 help='image files paths fed into network, single or multiple images')
+        self.parser.add_argument('--show_flg', default='false', type=str2bool, help='whether show the visualization result')
+        self.parser.add_argument('--dump_res', default='true', type=str2bool,
+                                 help='whether write out the visualization image')
+        self.parser.add_argument('--dump_vertex', default='false', type=str2bool,
+                                 help='whether write out the dense face vertices to mat')
+        self.parser.add_argument('--dump_ply', default='true', type=str2bool)
+        self.parser.add_argument('--dump_pts', default='true', type=str2bool)
+        self.parser.add_argument('--dump_roi_box', default='false', type=str2bool)
+        self.parser.add_argument('--dump_pose', default='true', type=str2bool)
+        self.parser.add_argument('--dump_depth', default='true', type=str2bool)
+        self.parser.add_argument('--dump_pncc', default='true', type=str2bool)
+        self.parser.add_argument('--dump_paf', default='true', type=str2bool)
+        self.parser.add_argument('--paf_size', default=3, type=int, help='PAF feature kernel size')
+        self.parser.add_argument('--dump_obj', default='true', type=str2bool)
+        self.parser.add_argument('--dlib_bbox', default='true', type=str2bool, help='whether use dlib to predict bbox')
+        self.parser.add_argument('--dlib_landmark', default='true', type=str2bool,
+                                 help='whether use dlib landmark to crop image')
+        self.parser.add_argument('-m', '--mode', default='gpu', type=str, help='gpu or cpu mode')
+        self.parser.add_argument('--bbox_init', default='two', type=str,
+                                 help='one|two: one-step bbox initialization or two-step')
+        self.parser.add_argument('--dump_2d_img', default='true', type=str2bool, help='whether to save 3d rendered image')
+        self.parser.add_argument('--dump_param', default='true', type=str2bool, help='whether to save param')
+        self.parser.add_argument('--dump_lmk', default='true', type=str2bool, help='whether to save landmarks')
+        self.parser.add_argument('--save_dir', default='./algorithm/DDFA/results', type=str, help='dir to save result')
+        self.parser.add_argument('--save_lmk_dir', default='./example', type=str, help='dir to save landmark result')
+        self.parser.add_argument('--img_list', default='./txt_name_batch.txt', type=str, help='test image list file')
+        self.parser.add_argument('--rank', default=0, type=int, help='used when parallel run')
+        self.parser.add_argument('--world_size', default=1, type=int, help='used when parallel run')
+        self.parser.add_argument('--resume_idx', default=0, type=int)
+        self.args = self.parser.parse_args()
+        self.synthetic = MultiviewDataGeneration(self.args)
```

Comment:
`str2bool` seems to be an unresolved reference, are you missing an import?

Also, is it actually needed to have all these arguments exposed to the end user to run this node successfully? OpeDR ROS nodes should provide simple examples of the toolkit's tools with a limited ability to change their behaviour via argparse for convenience .

It is quite confusing as to what this node is doing. A ROS subscription is created to get images from a webcam, but here a `path-in` argument is defined with a path to images on the disk. 

Correct me if i am wrong, but should this node work similarly to the human generation ROS nodes where a service is created that takes requests with images and publishes the result, as well as a client that uses the service as an example.

Reply:
Hi @tsampazk. The following modifications are done in order to be functional the node:
1)The str2bool import is added.
2)the extensive arguments are necessary in order to easily be executed the internally declarated class ''MultiviewDataGeneration(self.args)''.
3) the images are collected from the ros topic and are saved temporally in the disk. The idea of making the ros2 module in services might not work because we have to make extensive changes in the complicated structure of the original repo of the algorithm so it not might be functional at all. The issue is obviously based in implementation orientation of the original structure repository Rotate-and-Render.